### ISSUE TITLE
Data quota in app has no message

### ISSUE THREAD

**Issue by gh-andi** [owncloud/ios#822]
Hello Everyone,
Thanks providing this app 

I work at a company that is working with owncloud to set up an instance of owncloud.
We shortly before implemented the app in to this solution and are now testing it 

The idea we're working with is that all users have 0mb quota and the admin has unlimited quota.
The folder structure for all the users is created in the admin user root and layer by layer shared with the users that have access. The idea that users don't have any personal quota is that users should not use this solution as a dumping ground for personal holiday/cat pictures etc.

Users are not able to upload a file or picture or whatever to their root folder in the web version of the application. A message is displayed that user has no quota, this is expected behavior. But in the app the user is being stopped to upload a file in their root folder as well but no message is displayed. 

Thanks for your help in advance

### Steps to reproduce
1. Have an account that has a 0mb quota on it 
2. Go on the app and upload a picture to the root folder of the user 

### Expected behavior
The file is not uploaded and a message should pop up saying that the user has no quota left and/or that the shared folders from users with unlimited quota should be used (maybe custom message)

### Actual behavior
The file is not being uploaded but no message is displayed. In the uploads tab the file shows up but under it, the message "waiting on server connection/Warten auf Serververbindung(german)" is displayed

### Server configuration
**Web server:**
2 redhat webserver with owncloud version 9.1.3 on them

**Database:**
1 mariadb node 

**PHP version:**
both webservers run php version 7.0.2 according to the php -v command

### Client
**iOS version:**
Ios version 10.3 but happend on multiple versions

**ownCloud app version:**
3.5.2

**Device model:**
iphone 7 plus
but this bug appeared on all devices tested

#### ownCloud log (data/owncloud.log)
```
{"reqId":"WGOvwbiABIDWeWeGF-GI2wAAAAk","remoteAddr":"192.168.224.185","app":"webdav","message":"Exception: {\"Message\":\"HTTP\\\/1.1 507 Insufficient Storage\",\"Exception\":\"Sabre\\\\DAV\\\\Exception\\\\InsufficientStorage\",\"Code\":0,\"Trace\":\"#0 [internal function]: OCA\\\\DAV\\\\Connector\\\\Sabre\\\\QuotaPlugin->checkQuota('\\\/Photo-2016-12-...', Resource id #548, Object(OCA\\\\DAV\\\\Connector\\\\Sabre\\\\Directory), false)\\n#1 \\\/var\\\/www\\\/html\\\/owncloud-9.1.3\\\/3rdparty\\\/sabre\\\/event\\\/lib\\\/EventEmitterTrait.php(105): call_user_func_array(Array, Array)\\n#2 \\\/var\\\/www\\\/html\\\/owncloud-9.1.3\\\/3rdparty\\\/sabre\\\/dav\\\/lib\\\/DAV\\\/Server.php(1034): Sabre\\\\Event\\\\EventEmitter->emit('beforeCreateFil...', Array)\\n#3 \\\/var\\\/www\\\/html\\\/owncloud-9.1.3\\\/3rdparty\\\/sabre\\\/dav\\\/lib\\\/DAV\\\/CorePlugin.php(523): Sabre\\\\DAV\\\\Server->createFile('Photo-2016-12-2...', Resource id #548, NULL)\\n#4 [internal function]: Sabre\\\\DAV\\\\CorePlugin->httpPut(Object(Sabre\\\\HTTP\\\\Request), Object(Sabre\\\\HTTP\\\\Response))\\n#5 \\\/var\\\/www\\\/html\\\/owncloud-9.1.3\\\/3rdparty\\\/sabre\\\/event\\\/lib\\\/EventEmitterTrait.php(105): call_user_func_array(Array, Array)\\n#6 \\\/var\\\/www\\\/html\\\/owncloud-9.1.3\\\/3rdparty\\\/sabre\\\/dav\\\/lib\\\/DAV\\\/Server.php(459): Sabre\\\\Event\\\\EventEmitter->emit('method:PUT', Array)\\n#7 \\\/var\\\/www\\\/html\\\/owncloud-9.1.3\\\/3rdparty\\\/sabre\\\/dav\\\/lib\\\/DAV\\\/Server.php(248): Sabre\\\\DAV\\\\Server->invokeMethod(Object(Sabre\\\\HTTP\\\\Request), Object(Sabre\\\\HTTP\\\\Response))\\n#8 \\\/var\\\/www\\\/html\\\/owncloud-9.1.3\\\/apps\\\/dav\\\/appinfo\\\/v1\\\/webdav.php(56): Sabre\\\\DAV\\\\Server->exec()\\n#9 \\\/var\\\/www\\\/html\\\/owncloud-9.1.3\\\/remote.php(164): require_once('\\\/var\\\/www\\\/html\\\/o...')\\n#10 {main}\",\"File\":\"\\\/var\\\/www\\\/html\\\/owncloud-9.1.3\\\/apps\\\/dav\\\/lib\\\/Connector\\\/Sabre\\\/QuotaPlugin.php\",\"Line\":112,\"User\":\"scha.ch@ANYACCESS.NET\"}","level":4,"time":"2016-12-28T12:27:46+00:00","method":"PUT","url":"\/oc-shib\/remote.php\/webdav\/Photo-2016-12-28-12-07-02_0517.PNG","user":"scha.ch@ANYACCESS.NET"}


### Tell us what could be improved:
It would be nice to have a message notifying the user he has no quota
The icing on the cake then would be to be able to customize this message from an admin side

Thanks

**Comment by jesmrec** [comment]
This feature fits with the error handling, an important issue that will be handled for sure.

**Comment by jesmrec** [comment]
Approved

**Comment by nasli** [comment]
This will be within 3.5.3 release. 

Uploads will fail with not enough space error and moved to failed uploads queue. Tap on any upload with this kind of error will make a retry of all failed uploads with the same kind of error for the same account.
